module/behaviour/planning/SimpleWalkPathPlanner/src/SimpleWalkPathPlanner.cpp

```diff
@@ -144,97 +149,128 @@ namespace module::behaviour::planning {
+                        emit(std::make_unique<StopCommand>(subsumptionId));
+                        return;
                 }
+            });
 
+        // Save the plan cmd into latestCommand
+        on<Trigger<MotionCommand>, Sync<SimpleWalkPathPlanner>>().then(
+            [this](const MotionCommand& cmd) { latestCommand = cmd; });
+    }
 
-                float angle = std::atan2(position.y(), position.x()) + headingChange;
+    //-------- Add extra behaviours for path planning here and in the switch case of this file
 
-                angle = std::min(turnSpeed, std::max(angle, -turnSpeed));
+    void SimpleWalkPathPlanner::walkDirectly() {
+        emit(std::move(std::make_unique<WalkCommand>(subsumptionId, latestCommand.walk_command)));
+        emit(std::make_unique<ActionPriorities>(ActionPriorities{subsumptionId, {40, 11}}));
+    }
 
+    void SimpleWalkPathPlanner::determineSimpleWalkPath(const Ball& ball,
+                                                        const Field& field,
+                                                        const Sensors& sensors,
+                                                        const KickPlan& kickPlan,
+                                                        const FieldDescription& fieldDescription) {
 
-                // Euclidean distance to ball
-                float scaleF            = 2.0 / (1.0 + std::exp(-a * std::fabs(position.x()) + b)) - 1.0;
-                float scaleF2           = angle / M_PI;
-                float finalForwardSpeed = speedFactor * forwardSpeed * scaleF * (1.0 - scaleF2);
+        Eigen::Affine3d Htw(sensors.Htw);
 
-                float scaleS         = 2.0 / (1.0 + std::exp(-a * std::fabs(position.y()) + b)) - 1.0;
-                float scaleS2        = angle / M_PI;
-                float finalSideSpeed = -speedFactor * ((0.0 < position.y()) - (position.y() < 0.0)) * sideStep
-                                       * sideSpeed * scaleS * (1.0 - scaleS2);
+        auto now                = NUClear::clock::now();
+        float timeSinceBallSeen = std::chrono::duration_cast<std::chrono::nanoseconds>(now - timeBallLastSeen).count()
+                                  * (1.0f / std::nano::den);
 
 
-                std::unique_ptr<WalkCommand> command =
-                    std::make_unique<WalkCommand>(subsumptionId,
-                                                  Eigen::Vector3d(finalForwardSpeed, finalSideSpeed, angle));
+        Eigen::Vector3d rBWw_temp(ball.position.x(), ball.position.y(), fieldDescription.ball_radius);
 
-                emit(std::move(command));
-                emit(std::make_unique<ActionPriorities>(ActionPriorities{subsumptionId, {40, 11}}));
-            });
+        rBWw = timeSinceBallSeen < search_timeout ? rBWw_temp
+                                                  : Htw.inverse().linear().leftCols<1>() + Htw.inverse().translation();
```

Comment:
This avoid a common pitfall listed in the eigen docs [here](https://eigen.tuxfamily.org/dox/TopicPitfalls.html). (This will probably break formatting)
```suggestion
        if (timeSinceBallSeen < search_timeout) {
            rBWw = EigenVector3d(ball.position.x(), ball.position.y(), fieldDescription.ball_radius);
        }
        else {
            rBWw = Htw.inverse().linear().leftCols<1>() + Htw.inverse().translation();
        }
```